File: src/main/java/cz/astro/var/data/security/SecurityServiceImpl.java
package cz.astro.var.data.security;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Component;

@Component
public class SecurityServiceImpl implements SecurityService {

    @Autowired
    private AuthenticationManager authenticationManager;

    @Autowired
    private JwtTokenProvider tokenProvider;

    @Override
    public UserPrincipal getCurrentUser() {
        if (SecurityContextHolder.getContext().getAuthentication() != null) {
            return (UserPrincipal) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
        }
        return null;
    }

    @Override
    public JwtAuthResponse loginUser(JwtAuthRequest user) {
        Authentication authentication = authenticationManager.authenticate(
                new UsernamePasswordAuthenticationToken(
                        user.getUsername(),
                        user.getPassword()
                )
        );

        SecurityContextHolder.getContext().setAuthentication(authentication);

        String jwt = tokenProvider.generateToken(authentication);

        return new JwtAuthResponse(jwt);
    }
}

File: backend/src/main/kotlin/cz/astro/var/data/security/SecurityServiceImpl.kt
package cz.astro.`var`.data.security

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.Authentication
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component

@Component
class SecurityServiceImpl : SecurityService {

    @Autowired
    private val authenticationManager: AuthenticationManager? = null

    @Autowired
    private val tokenProvider: JwtTokenProvider? = null

    override fun getCurrentUser(): UserPrincipal? {
        return if (SecurityContextHolder.getContext().authentication != null) {
            SecurityContextHolder.getContext().authentication.principal as UserPrincipal
        } else null
    }

    override fun loginUser(user: JwtAuthRequest): JwtAuthResponse {
        val authentication = authenticationManager!!.authenticate(
                UsernamePasswordAuthenticationToken(
                        user.username,
                        user.password
                )
        )

        SecurityContextHolder.getContext().authentication = authentication

        val jwt = tokenProvider!!.generateToken(authentication)

        return JwtAuthResponse(jwt)
    }
}
